src/material/core/style/_sass-utils.scss

```diff
@@ -67,9 +68,18 @@ $use-system-typography-variables: false;
   @if (meta.type-of($color) == 'color') {
     @return color.change($color, $args...);
   }
-  @else if ($color != null and map.get($args, alpha) != null and $use-system-color-variables) {
+  @else if ($color != null and map.get($args, alpha) != null) {
     $opacity: map.get($args, alpha);
-    @return #{color-mix(in srgb, #{$color} #{($opacity * 100) + '%'}, transparent)};
+    @if meta.type-of($opacity) == number {
+      $opacity: ($opacity * 100) + '%';
+    }
+
+    // Wrap the $color as a `var()` if it's the name of a CSS variable
+    @if (meta.type-of($opacity) == string and string.index($color, '--') == 1) {
```

Comment:
Nit: I think it might make sense to extract this check into our inspection api since it is already used twice in this PR and will probably start getting used more and more

Reply:
Good call - added a `sass-util` for checking this, and added it where this conditional was used